requests/utils.py

```diff
@@ -87,7 +83,22 @@ def super_len(o):
             # is actually a special file descriptor like stdin. In this
             # instance, we don't know what the length is, so set it to zero and
             # let requests chunk it instead.
-            current_position = total_length
+            if total_length is not None:
+                current_position = total_length
+
+        if hasattr(o, 'seek') and total_length is None:
```

Comment:
So I like that this as a simpler option to the two hasattr checks, but we'd still have to leave those in place for StringIO-like objects, so it would end up being a somewhat redundant portion of the conditional.

I do, however, think we're likely going to want to extract [this logic](https://github.com/nateprewitt/requests/blob/f5c59743e7a45b68eaa82c83baaaed5607348afe/requests/utils.py#L78-L98) into it's own function in order to solve #3079 so maybe it could fit in with that somehow? I haven't worked up a good solution for that yet though.

@Lukasa, do you have any opinions on using this?